src/index.tsx

```diff
@@ -46,26 +56,154 @@ function Updaters() {
+  const walletOverrideBackfilled = useAppSelector((state) => state.user.walletOverrideBackfilled)
+
+  const injectedIsActive = hooks.useSelectedIsActive(injected)
+  const coinbaseWalletIsActive = hooks.useSelectedIsActive(coinbaseWallet)
+  const walletConnectIsActive = hooks.useSelectedIsActive(walletConnect)
+
+  const previousInjectedIsActive = usePrevious(injectedIsActive)
+  const previousCoinbaseWalletIsActive = usePrevious(coinbaseWalletIsActive)
+  const previousWalletConnectIsActive = usePrevious(walletConnectIsActive)
+
+  const injectedIsActivating = hooks.useSelectedIsActivating(injected)
+  const coinbaseWalletIsActivating = hooks.useSelectedIsActivating(coinbaseWallet)
+  const walletConnectIsActivating = hooks.useSelectedIsActivating(walletConnect)
+
+  const previousInjectedIsActivating = usePrevious(injectedIsActivating)
+  const previousCoinbaseWalletIsActivating = usePrevious(coinbaseWalletIsActivating)
+  const previousWalletConnectIsActivating = usePrevious(walletConnectIsActivating)
+
+  const [isInjectedEagerlyConnecting, setIsInjectedEagerlyConnecting] = useReducer(
+    (_state: boolean, action: boolean) => action,
+    false
+  )
+  const [isCoinbaseWalletEagerlyConnecting, setIsCoinbaseWalletEagerlyConnecting] = useReducer(
+    (_state: boolean, action: boolean) => action,
+    false
+  )
+  const [isWalletConnectEagerlyConnecting, setIsWalletConnectEagerlyConnecting] = useReducer(
+    (_state: boolean, action: boolean) => action,
+    false
+  )
+
+  useEffect(() => {
+    const injectedState: ConnectorState = {
+      isActive: injectedIsActive,
+      previousIsActive: previousInjectedIsActive,
+      isActivating: injectedIsActivating,
+      previousIsActivating: previousInjectedIsActivating,
+      isEagerlyConnecting: isInjectedEagerlyConnecting,
+      setIsEagerlyConnecting: setIsInjectedEagerlyConnecting,
+    }
+    const coinbaseWalletState: ConnectorState = {
+      isActive: coinbaseWalletIsActive,
+      previousIsActive: previousCoinbaseWalletIsActive,
+      isActivating: coinbaseWalletIsActivating,
+      previousIsActivating: previousCoinbaseWalletIsActivating,
+      isEagerlyConnecting: isCoinbaseWalletEagerlyConnecting,
+      setIsEagerlyConnecting: setIsCoinbaseWalletEagerlyConnecting,
+    }
+    const walletConnectState: ConnectorState = {
+      isActive: walletConnectIsActive,
+      previousIsActive: previousWalletConnectIsActive,
+      isActivating: walletConnectIsActivating,
+      previousIsActivating: previousWalletConnectIsActivating,
+      isEagerlyConnecting: isWalletConnectEagerlyConnecting,
+      setIsEagerlyConnecting: setIsWalletConnectEagerlyConnecting,
+    }
+    const isActiveMap = new Map<Wallet, ConnectorState>([
+      [Wallet.INJECTED, injectedState],
+      [Wallet.COINBASE_WALLET, coinbaseWalletState],
+      [Wallet.WALLET_CONNECT, walletConnectState],
+    ])
+
+    isActiveMap.forEach((state: ConnectorState, wallet: Wallet) => {
```

Comment:
I'm unsure why the previous states are relevant. Can you leave some comments explaining this logic?